cardano-testnet/src/Util/Process.hs

```diff
@@ -132,25 +131,11 @@ assertByDeadlineIOCustom str deadline f = GHC.withFrozenCallStack $ do
   success <- liftIO f
   unless success $ do
     currentTime <- liftIO DTC.getCurrentTime
+    note_ $ "Current time: " <> show currentTime
     if currentTime < deadline
       then do
         liftIO $ IO.threadDelay 1000000
         assertByDeadlineIOCustom str deadline f
       else do
         H.annotateShow currentTime
         failMessage GHC.callStack $ "Condition not met by deadline: " <> str
-
-assertByDeadlineMCustom
-  :: (MonadTest m, MonadIO m, HasCallStack)
-  => String -> UTCTime -> m Bool -> m ()
-assertByDeadlineMCustom str deadline f = GHC.withFrozenCallStack $ do
-  success <- f
-  unless success $ do
-    currentTime <- liftIO DTC.getCurrentTime
-    if currentTime < deadline
-      then do
-        liftIO $ IO.threadDelay 1000000
-        assertByDeadlineMCustom str deadline f
-      else do
-        H.annotateShow currentTime
-        failMessage GHC.callStack $ "Condition not met by deadline: " <> str
```

Comment:
Replace `assertByDeadlineMCustom` with `byDeadlineM`.  The latter is better because it catches assertions and retries whereas an assertion in the block of the former will fail the test.

Reply:
`byDeadlineM` will also take a poll interval so we don't needlessly poll too often which clutters the output.